resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -502,6 +521,25 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
     )
   }
 
+  private def satisfiesLocality(offerHostname: String): Boolean = {
+    if (hostToLocalTaskCount.nonEmpty) {
```

Comment:
I see what you mean. I wanted these semantics to only be in-effect if location information is available. If there is no location information, I don't think there is a reason to wait for a delay.

So, the original semantics I wanted was, "If location information is available, launch an executor on a host only when we have a task that wants to be on that host, or the configurable delay has elapsed."

What should the semantics be?